README.md

```diff
@@ -173,3 +173,4 @@ package doesn't currently support the functionality that you need.
 # Gotchas
 
 1. `Integer`s typically don't take part in the kind of optimisation procedures that this package is designed to handle. Consequently, `flatten(::Integer)` produces an empty vector.
+2. `deferred` has some type-stability issues when used in conjunction with abstract types. For example, `flatten(deferred(Normal, 5.0, 4.0))` won't infer properly. A simple work around is to write a function `normal(args...) = Normal(args...)` and work with `deferred(normal, 5.0, 4.0)` instead.
```

Comment:
Hmm, this is a bit awkward :(

Reply:
Yeah, it's a bit sad. I'm not really sure what to do about it for now, so I figure it's better to document the problem and the workaround until we find a nice solution.

Reply:
Out of curiosity, do you know if this holds true for even Julia nightly? I'm just wondering if we should open an issue against the base repo to try and handle this case?